modules/phase_field/src/materials/DerivativeParsedMaterialHelper.C

```diff
@@ -154,6 +157,9 @@ DerivativeParsedMaterialHelper::assembleDerivatives()
     // remove the 'current' element from the queue
     queue.pop_front();
   }
+  // cleanup zero derivatives
+  for (std::vector<ADFunction*>::iterator it = adfunctions.begin(); it != adfunctions.end(); ++it)
+    delete *it;
```

Comment:
On second thought I think I'd rather change all of this to use `MooseSharedPointers` internally. That makes the code shorter by having automatic cleanup.